### ISSUE TITLE
Repository creation for org members is enabled by default and leads to permission inconsistencies

### ISSUE THREAD

**Issue by fredg02** [EclipseFdn/eclipsefdn-github-sync#51]
Repository creation for organization members seems to be enabled by default, when a new org is created. Here are a few examples
* https://github.com/organizations/eclipse-iofog/settings/member_privileges
* https://github.com/organizations/eclipse-openj9/settings/member_privileges
* https://github.com/organizations/eclipse-embed-cdt/settings/member_privileges
* https://github.com/organizations/eclipse-emfcloud/settings/member_privileges
* https://github.com/organizations/eclipse-glsp/settings/member_privileges

When members create repos, they are not added in PMI and therefore are not considered during a GitHub sync script run. This leads to missing teams (<project-name>-committers, <project-name>-contributors, <project-name>-project-leads) and committers having admin rights on projects (since they have created them).

If possible, we should check that this option is disabled and disable it if it's not through the API.

Related question: What do we do with existing repos that were created by members? Do we let the project leads confirm that they are legit and add them to the PMI manually?

**Comment by autumnfound** [comment]
So the request here is to ensure that settings that are desired in orgs are enforced on a regular basis through the script? Are there other settings that we want to manage while we're posting the update?

https://octokit.github.io/rest.js/v17#orgs-update

**Comment by fredg02** [comment]
At a first glance, I'd say the following options should be checked and set:

* default_repository_permission = read
* members_can_create_repositories = false
* members_can_create_private_repositories = false
* members_can_create_public_repositories = false
* members_allowed_repository_creation_type = none

**Comment by chrisguindon** [comment]
> If possible, we should check that this option is disabled and disable it if it's not through the API.

@fredg02 can you or someone in Eclipse webmaster review all our orgs and make this that this option is disabled. Please ignore if you already done that!

We do have plans for managing important settings via the API. Right now, we are using https://github.com/EclipseFdn/eclipsefdn-github-sync/issues/34 to track the requirements. My hope is that we will have time to fix this in the futur.

@fredg02  Can you post your comment there a?

**Comment by fredg02** [comment]
> > If possible, we should check that this option is disabled and disable it if it's not through the API.
> 
> @fredg02 can you or someone in Eclipse webmaster review all our orgs and make this that this option is disabled. Please ignore if you already done that!

I hope we can invest the time and effort into automating this instead of checking 30+ orgs manually.

> We do have plans for managing important settings via the API. Right now, we are using #34 to track the requirements. My hope is that we will have time to fix this in the futur.
> 
> @fredg02 Can you post your comment there a[s well]?

Yes.

**Comment by chrisguindon** [comment]
> I hope we can invest the time and effort into automating this instead of checking 30+ orgs manually.

It will take some time but now that we know of the issue, we need to fix it! Can webmaster review all the orgs? 

I am asking webmaster since it's the account responsible for creating the orgs.

**Comment by fredg02** [comment]
> It will take some time but now that we know of the issue, we need to fix it! Can webmaster review all the orgs?

I expect that it will take as much time to review all the orgs manually as implementing the tooling to fix it automatically. Hence, I'd recommend to fix it in the sync script first. Let me know if I completely underestimate the effort.

**Comment by fredg02** [comment]
> Related question: What do we do with existing repos that were created by members? Do we let the project leads confirm that they are legit and add them to the PMI manually?

@waynebeaton: can you comment on that question?

**Comment by waynebeaton** [comment]
> > Related question: What do we do with existing repos that were created by members? Do we let the project leads confirm that they are legit and add them to the PMI manually?
> 
> @waynebeaton: can you comment on that question?

It depends on the nature of the IP contained in those repos. If the IP was developed by the project team (implying that the content has been developed since the project became an Eclipse project), under the terms of the EDP, following the Eclipse IP Policy, then yes. At a minimum, we'd need a documented personal assertion from the project lead that this is the case.

Otherwise, no.

**Comment by fredg02** [comment]
> It depends on the nature of the IP contained in those repos. If the IP was developed by the project team (implying that the content has been developed since the project became an Eclipse project), under the terms of the EDP, following the Eclipse IP Policy, then yes. At a minimum, we'd need a documented personal assertion from the project lead that this is the case.
> 
> Otherwise, no.
@waynebeaton OK. How do we get this process started for the repos in the eclipse-iofog organization?

**Comment by autumnfound** [comment]
> > It will take some time but now that we know of the issue, we need to fix it! Can webmaster review all the orgs?
> 
> I expect that it will take as much time to review all the orgs manually as implementing the tooling to fix it automatically. Hence, I'd recommend to fix it in the sync script first. Let me know if I completely underestimate the effort.

The reason we want to review orgs before the fix is to ensure that we aren't breaking projects unduly. There have been exceptions for things in the past, and I think it would be a good idea to be aware of potential stumbling blocks before we hit them.

**Comment by chrisguindon** [comment]
> I expect that it will take as much time to review all the orgs manually as implementing the tooling to fix it automatically. Hence, I'd recommend to fix it in the sync script first. Let me know if I completely underestimate the effort.

We estimate 0.5 to 1 day of work with testing. We are adding this task to our queue. 

My hope is that we can publish a fix to production by the end of next week. 

I will defer to you about fixing this manually in the meantime or if you are ok to wait until this patch is published.

**Comment by waynebeaton** [comment]
> @waynebeaton OK. How do we get this process started for the repos in the eclipse-iofog organization?

Were these created before or after the project moved to the Eclipse Foundation?

**Comment by fredg02** [comment]
> The reason we want to review orgs before the fix is to ensure that we aren't breaking projects unduly. There have been exceptions for things in the past, and I think it would be a good idea to be aware of potential stumbling blocks before we hit them.

Good point. We should handle that with a dry run. First, check which repos have the option set and create a list. Then discuss possible exceptions, etc. and finally a full run.

**Comment by fredg02** [comment]
> Were these created before or after the project moved to the Eclipse Foundation?

@waynebeaton: According to the audit log, the following projects were created after the project moved to the Eclipse Foundation:

action | actor | org | repo | date
-- | -- | -- | -- | --
repo.create | committer | eclipse-iofog | eclipse-iofog/helm | 2019-04-22
repo.create | committer | eclipse-iofog | eclipse-iofog/test-runner | 2019-04-29
repo.create | committer | eclipse-iofog | eclipse-iofog/cli | 2019-05-23
repo.create | committer | eclipse-iofog | eclipse-iofog/homebrew-iofogctl | 2019-06-11
repo.create | committer | eclipse-iofog | eclipse-iofog/ECN-Viewer | 2019-07-19
repo.create | committer | eclipse-iofog | eclipse-iofog/script | 2019-11-12
repo.create | committer | eclipse-iofog | eclipse-iofog/octant-plugin | 2019-11-13
repo.create | committer | eclipse-iofog | eclipse-iofog/port-manager | 2020-01-14
repo.create | project lead | eclipse-iofog | eclipse-iofog/documentation | 2020-01-11
repo.create | committer | eclipse-iofog | eclipse-iofog/skupper-proxy | 2020-01-27
repo.create | committer | eclipse-iofog | eclipse-iofog/integrations | 2020-01-27
repo.create | committer | eclipse-iofog | eclipse-iofog/router | 2020-03-14
repo.create | committer | eclipse-iofog | eclipse-iofog/Agent-Go | 2020-04-21

**Comment by waynebeaton** [comment]
My best guess is we can consider them all project code written under the supervision of the PMC and just add 'em to the PMI.

Is anybody in active contact with the project team? Is there a bug open specifically addressing the eclipse-iofog organization? I'd still like the project team's acknowledgement that these repositories and all of their contents have been created under the terms of the EDP, that all contributors have ECAs, etc.

It's likely that they didn't know that they weren't supposed to be able to create repositories, and assumed that what the permissions model let them do aligned with our processes.

**Comment by fredg02** [comment]
> Is anybody in active contact with the project team?

No

> Is there a bug open specifically addressing the eclipse-iofog organization? 

I stumbled over the inconsistencies while working on https://bugs.eclipse.org/bugs/show_bug.cgi?id=562627, but I guess a separate bug is in order.

**Comment by fredg02** [comment]
@autumnfound can you generate a list of organization that currently have the repository creation enabled for members and corresponding repos that have been created with this permission (tell-tale would be missing committers-, contributors-, project lead- groups and/or a committer with admin permissions) so that EMO can get in contact with the projects?

**Comment by autumnfound** [comment]
I've started work on the report, and hope to have the report finished before 12pm EST tomorrow latest. Once it is complete I will post a zipped CSV to this issue

**Comment by autumnfound** [comment]
This will take longer than expected as there is a lot of functionality that does not exist in our wrapper yet and needs to be created. This should be ready for next Tuesday (as Monday is a holiday in Canada). Sorry for the delay!

**Comment by autumnfound** [comment]
Update: The code is complete and the report can be run once we have access to the webmaster access token. Write access is required to read collaborators for some reason.

**Comment by autumnfound** [comment]
These changes have been made live
